backend/src/routes/api/notebooks/index.ts

```diff
@@ -49,7 +51,22 @@ module.exports = async (fastify: KubeFastifyInstance) => {
           ? false
           : await getNotebookStatus(fastify, namespace, name);
 
-        return { notebook, isRunning };
+        const notebookName = notebook?.metadata.name;
+        let newNotebook: Notebook | void;
+        if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
+          const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
+            fastify.log.warn(`Failed patching route to notebook ${notebookName}, ${e.message}`);
+          });
+          if (route) {
+            newNotebook = await patchNotebookRoute(fastify, route, namespace, notebookName).catch(
```

Comment:
nit/opinion: this is a side effect in a `get` function, which is a bit counter intuitive ...

Shouldn't this patching operation be done by the `odh-notebook-controller`?

Reply:
This is a fallback if we failed to get it at creation time (which we also improved beyond the 500ms wait). This code shouldn't actually run -- the if-statement above should stop it all -- _all the time_. I get that the side effect in a get call is weird, but I am not going to request a change here.

This call gets called in a few spots where if the user somehow didn't get the link, they'll get it the next time they login and try to use their notebook. Unless I hear a good alternative for a safety fallback, this is what we will have.

Reply:
ack for not changing it here, I understand the logic

what about doing it in `odh-notebook-controller`, when the route is being created? could you point me to where this route is actually created?
(I'm not pushing for any change, just trying to better understand the architecture and the responsibilities:)

Reply:
@kpouget I don't have that information and I am not sure I could find it in a timely manner. Perhaps check with Chris. It is not within' the ODH Dashboard repo. I suspect it is in the Kubeflow github org of repos.